### DISCUSSION TITLE
Incoming transfer event?

### DISCUSSION THREAD

**Discussion by benams** [ConsenSys-archive/ganache#4096] [Q&A]
Can I subscribe to an event when ETH is transferred to an account? 

For example: I simulate a transaction from account X, to smart contract S, that will send some ETH to accounts A-D. 
I want to get notified whenever the ETH is transferred to accounts A-D. 

EDIT: I can see there is a "built in block explorer" for Ganache UI, I guess it means I can somehow touch it with the cli somehow. Maybe the solution is there. Can someone share how we can use this explorer please?

**Comment by davidmurdoch** [comment]
There isn't an easy way to do this with only Ganache, as it's pretty low level. You could use the ganache npm package programmatically with JavaScript to listen for specific opcodes that transfer value (like CALL) via the `ganache:vm:tx:step` event. See https://github.com/trufflesuite/ganache#ganache-provider-events

**Reply by benams** [reply]
Thanks for your reply. 
Any documentation explains the opcodes(and the rest of the event object)?

**Reply by davidmurdoch** [reply]
You can learn more about the individual opcodes at https://www.evm.codes/

The event ganache emits is derived from: https://github.com/ethereumjs/ethereumjs-monorepo/blob/696b36fe9091cd67f3e0a70bc696e41bd16e57a2/packages/evm/docs/interfaces/InterpreterStep.md You may need to dive into their source code a bit to fully understand each property.
